DOCS-211 Data transfer/migration tutorial draft (#59)
hazelcast/cloud-docs: docs/modules/ROOT/pages/data-migration.adoc

= Transfer Data to Another Cluster
:description: In this tutorial, you'll transfer data from one cluster to another using xref:wan-replication.adoc[WAN Replication]. You'll also learn how to safely delete a cluster after the successful transfer of data.

{description}

== Context

WAN replication supports the following use cases:

* Migrate data to a new cluster before decommissioning your original cluster.

* Use WAN replication with a xref:failover-clients-with-hazelcast-cloud.adoc[failover client] as part of your disaster recovery strategy. WAN replication automatically synchronizes data between clusters. If a cluster fails, the failover client can switch to an alternative cluster with minimal interruption to service.

== Before you Begin

> * Set up two active clusters with developer mode turned off. Use the same configuration for each cluster. For example, if one cluster

Reviewer: No need to qualify the clusters as active here. Active is just a state that means the cluster is running. If a user has set up the clusters, we can assume it's running.

```suggestion
* Set up two new clusters. You can use either a xref:create-serverless-cluster.adoc[Serverless production cluster] or a xref:create-dedicated-cluster.adoc[Dedicated cluster]. Make sure that each cluster is configured with the same settings. For example, if one cluster
```

Author: Resolved in the latest commit.

> contains xref:custom-classes-upload.adoc[custom classes] or persistence,

Reviewer: ```suggestion
contains xref:cluster-side-modules.adoc[cluster-side modules]
```

Author: Resolved in the latest commit.

> enable and configure the same options for the other cluster.

Reviewer: ```suggestion
make sure that the other cluster has the same ones.
```

Author: Resolved in the latest commit, although slight rewording.

* Decide which cluster you want to:

> ** Send or migrate data from. This is the _publisher_ cluster.

Reviewer: It's confusing terminology, but we use 'active' and 'passive' to describe WAN clusters. This is slightly confusing because 'active' clusters can also be clusters in an active state.

Active-Passive: An active cluster replicates its maps and/or caches to one or more passive clusters for the purpose of maintaining a backup. If the active cluster becomes unavailable, you can redirect the application traffic to the passive cluster.

I prefer your terms. Maybe let's say 'also known as...'?

Author: Updated as suggested.

** Receive data. This is the _consumer_ cluster.

== Step 1. Set Up WAN Replication

Set up WAN replication on the publisher cluster to send data to your consumer cluster.

. Go to *Your Clusters* and select your publisher cluster.

> . Select *MANAGE CLUSTER > SETTINGS*.

Reviewer: Even if the UI text is in all caps, it's difficult to read. We used to use all caps in docs to be consistent with the UI, but now we go with title case.

Author: Resolved in the latest commit.

. On the *Cluster Settings* page, select *WAN Replication* and then click *+ CREATE NEW REPLICATION*.
+
The publisher cluster is set as the *Active* cluster. Any data updates performed on this cluster are automatically copied to the other cluster connected by the WAN replication. In this case, your consumer cluster.
+

> . Update the xref:wan-replication.adoc#creating-wan-replication[other parameters] as required.

Reviewer: Will you provide values for the user to input and explain them rather than linking? Tutorials should be self-contained. We shouldn't make the user go somewhere else and get lost.

Author: I've decided to leave the default values and provide a link in 'Learn more'/related links at the bottom of the page. Is there anyone I can check with to see what settings are recommended for the two use cases.

Author: Full description of parameters and values has now been added.

. Click *NEXT >* to configure the consumer cluster.
. From the *Cluster 2* list, select your consumer cluster.
. From the *Active/Passive* list, select *Passive* to ensure that your consumer cluster only receives data.
+ 
If you leave the setting as *Active*, any data updates performed on this cluster are automatically copied to your publisher cluster.
+   
. Click *Save*. WAN Replication is set up.

> . If you want to synchronize all data between the two clusters, click *Sync*. This option is useful if you intend to decommission the publisher cluster.

Reviewer: Maybe 'shut down' is better than decommission?

Author: Updated as suggested.

== Step 2. Verify WAN Replication

In this step, you'll check that data updates are copied from the publisher cluster to your consumer cluster.

> . Connect a client to your publisher cluster and write some data to a map. You can skip this step if you've already synched data.

Reviewer: We don't want users to skip this step. The tutorial starts from two newly set up clusters. So we should assume that environment.

Author: Updated as suggested.

> . Open the Management Center for your consumer cluster. From the dashboard of your consumer cluster, select *MANAGE CLUSTER > MANAGEMENT CENTER*.

Reviewer: Just a convention. We don't use an article with Management Center unless its qualifying something. For example 'in the Management Center UI...'

```suggestion
. Open Management Center for your consumer cluster. From the dashboard of your consumer cluster, select *MANAGE CLUSTER > MANAGEMENT CENTER*.
```

Author: Updated as suggested.

> = Transfer Data to Another Cluster

Reviewer: This page can be deleted.
